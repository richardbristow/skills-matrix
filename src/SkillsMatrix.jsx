import React from 'react';
import { Container } from 'react-bootstrap';

const SkillsMatrix = () => (
  <Container>
    <h1>Skills Matrix</h1>
  </Container>
);

export default SkillsMatrix;
