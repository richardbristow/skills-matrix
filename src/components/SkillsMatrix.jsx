import React from 'react';
import TopNavBar from './TopNavBar';

const SkillsMatrix = () => <TopNavBar />;

export default SkillsMatrix;
