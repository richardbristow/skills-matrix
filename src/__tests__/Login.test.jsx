import React from 'react';
import ReactDOM from 'react-dom';
import SkillsMatrix from '../SkillsMatrix';

describe('SkillsMatrix', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<SkillsMatrix />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});
