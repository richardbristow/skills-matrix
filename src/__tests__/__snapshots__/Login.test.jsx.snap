// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render the login form without throwing an error 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef(Bootstrap(FormGroup))
          controlId="email"
        >
          <ForwardRef(ContextTransform)>
            Email Address
          </ForwardRef(ContextTransform)>
          <ForwardRef(ContextTransform)
            autoFocus={true}
            onChange={[Function]}
            placeholder="Email address..."
            required={true}
            type="email"
            value=""
          />
          <Feedback
            as="div"
            type="invalid"
          >
            Please provide a valid email address.
          </Feedback>
        </ForwardRef(Bootstrap(FormGroup))>,
        <ForwardRef(Bootstrap(FormGroup))
          controlId="password"
        >
          <ForwardRef(ContextTransform)>
            Password
          </ForwardRef(ContextTransform)>
          <ForwardRef(ContextTransform)
            onChange={[Function]}
            placeholder="Password..."
            required={true}
            type="password"
            value=""
          />
          <Feedback
            as="div"
            type="invalid"
          >
            Please provide a password.
          </Feedback>
        </ForwardRef(Bootstrap(FormGroup))>,
        <ForwardRef(Bootstrap(Button))
          disabled={true}
          type="submit"
        >
          Login
        </ForwardRef(Bootstrap(Button))>,
      ],
      "noValidate": true,
      "onSubmit": [Function],
      "validated": false,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ForwardRef(ContextTransform)>
              Email Address
            </ForwardRef(ContextTransform)>,
            <ForwardRef(ContextTransform)
              autoFocus={true}
              onChange={[Function]}
              placeholder="Email address..."
              required={true}
              type="email"
              value=""
            />,
            <Feedback
              as="div"
              type="invalid"
            >
              Please provide a valid email address.
            </Feedback>,
          ],
          "controlId": "email",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Email Address",
            },
            "ref": null,
            "rendered": "Email Address",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": undefined,
              "propTypes": undefined,
              "render": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "autoFocus": true,
              "onChange": [Function],
              "placeholder": "Email address...",
              "required": true,
              "type": "email",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "Feedback": [Function],
              "defaultProps": undefined,
              "propTypes": undefined,
              "render": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": "div",
              "children": "Please provide a valid email address.",
              "type": "invalid",
            },
            "ref": null,
            "rendered": "Please provide a valid email address.",
            "type": [Function],
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "propTypes": undefined,
          "render": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ForwardRef(ContextTransform)>
              Password
            </ForwardRef(ContextTransform)>,
            <ForwardRef(ContextTransform)
              onChange={[Function]}
              placeholder="Password..."
              required={true}
              type="password"
              value=""
            />,
            <Feedback
              as="div"
              type="invalid"
            >
              Please provide a password.
            </Feedback>,
          ],
          "controlId": "password",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Password",
            },
            "ref": null,
            "rendered": "Password",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": undefined,
              "propTypes": undefined,
              "render": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onChange": [Function],
              "placeholder": "Password...",
              "required": true,
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "Feedback": [Function],
              "defaultProps": undefined,
              "propTypes": undefined,
              "render": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": "div",
              "children": "Please provide a password.",
              "type": "invalid",
            },
            "ref": null,
            "rendered": "Please provide a password.",
            "type": [Function],
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "propTypes": undefined,
          "render": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Login",
          "disabled": true,
          "type": "submit",
        },
        "ref": null,
        "rendered": "Login",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "propTypes": undefined,
          "render": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "Check": Object {
        "$$typeof": Symbol(react.forward_ref),
        "Input": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "propTypes": undefined,
          "render": [Function],
        },
        "Label": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "propTypes": undefined,
          "render": [Function],
        },
        "defaultProps": undefined,
        "propTypes": undefined,
        "render": [Function],
      },
      "Control": Object {
        "$$typeof": Symbol(react.forward_ref),
        "Feedback": [Function],
        "defaultProps": undefined,
        "propTypes": undefined,
        "render": [Function],
      },
      "Group": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "propTypes": undefined,
        "render": [Function],
      },
      "Label": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "propTypes": undefined,
        "render": [Function],
      },
      "Row": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "propTypes": undefined,
        "render": [Function],
      },
      "Text": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "propTypes": undefined,
        "render": [Function],
      },
      "defaultProps": undefined,
      "propTypes": undefined,
      "render": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef(Bootstrap(FormGroup))
            controlId="email"
          >
            <ForwardRef(ContextTransform)>
              Email Address
            </ForwardRef(ContextTransform)>
            <ForwardRef(ContextTransform)
              autoFocus={true}
              onChange={[Function]}
              placeholder="Email address..."
              required={true}
              type="email"
              value=""
            />
            <Feedback
              as="div"
              type="invalid"
            >
              Please provide a valid email address.
            </Feedback>
          </ForwardRef(Bootstrap(FormGroup))>,
          <ForwardRef(Bootstrap(FormGroup))
            controlId="password"
          >
            <ForwardRef(ContextTransform)>
              Password
            </ForwardRef(ContextTransform)>
            <ForwardRef(ContextTransform)
              onChange={[Function]}
              placeholder="Password..."
              required={true}
              type="password"
              value=""
            />
            <Feedback
              as="div"
              type="invalid"
            >
              Please provide a password.
            </Feedback>
          </ForwardRef(Bootstrap(FormGroup))>,
          <ForwardRef(Bootstrap(Button))
            disabled={true}
            type="submit"
          >
            Login
          </ForwardRef(Bootstrap(Button))>,
        ],
        "noValidate": true,
        "onSubmit": [Function],
        "validated": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ForwardRef(ContextTransform)>
                Email Address
              </ForwardRef(ContextTransform)>,
              <ForwardRef(ContextTransform)
                autoFocus={true}
                onChange={[Function]}
                placeholder="Email address..."
                required={true}
                type="email"
                value=""
              />,
              <Feedback
                as="div"
                type="invalid"
              >
                Please provide a valid email address.
              </Feedback>,
            ],
            "controlId": "email",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Email Address",
              },
              "ref": null,
              "rendered": "Email Address",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": undefined,
                "propTypes": undefined,
                "render": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "autoFocus": true,
                "onChange": [Function],
                "placeholder": "Email address...",
                "required": true,
                "type": "email",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "Feedback": [Function],
                "defaultProps": undefined,
                "propTypes": undefined,
                "render": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "div",
                "children": "Please provide a valid email address.",
                "type": "invalid",
              },
              "ref": null,
              "rendered": "Please provide a valid email address.",
              "type": [Function],
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": undefined,
            "propTypes": undefined,
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ForwardRef(ContextTransform)>
                Password
              </ForwardRef(ContextTransform)>,
              <ForwardRef(ContextTransform)
                onChange={[Function]}
                placeholder="Password..."
                required={true}
                type="password"
                value=""
              />,
              <Feedback
                as="div"
                type="invalid"
              >
                Please provide a password.
              </Feedback>,
            ],
            "controlId": "password",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Password",
              },
              "ref": null,
              "rendered": "Password",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": undefined,
                "propTypes": undefined,
                "render": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onChange": [Function],
                "placeholder": "Password...",
                "required": true,
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "Feedback": [Function],
                "defaultProps": undefined,
                "propTypes": undefined,
                "render": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "div",
                "children": "Please provide a password.",
                "type": "invalid",
              },
              "ref": null,
              "rendered": "Please provide a password.",
              "type": [Function],
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": undefined,
            "propTypes": undefined,
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Login",
            "disabled": true,
            "type": "submit",
          },
          "ref": null,
          "rendered": "Login",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": undefined,
            "propTypes": undefined,
            "render": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "Check": Object {
          "$$typeof": Symbol(react.forward_ref),
          "Input": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": undefined,
            "propTypes": undefined,
            "render": [Function],
          },
          "Label": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": undefined,
            "propTypes": undefined,
            "render": [Function],
          },
          "defaultProps": undefined,
          "propTypes": undefined,
          "render": [Function],
        },
        "Control": Object {
          "$$typeof": Symbol(react.forward_ref),
          "Feedback": [Function],
          "defaultProps": undefined,
          "propTypes": undefined,
          "render": [Function],
        },
        "Group": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "propTypes": undefined,
          "render": [Function],
        },
        "Label": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "propTypes": undefined,
          "render": [Function],
        },
        "Row": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "propTypes": undefined,
          "render": [Function],
        },
        "Text": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "propTypes": undefined,
          "render": [Function],
        },
        "defaultProps": undefined,
        "propTypes": undefined,
        "render": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
